### PR TITLE
Fixed LinkedIn to work with New Layout

### PR DESCRIPTION
**Pull request by eengoron**
Hey Phil and Company -- 

I fixed the lead clipper to work with the new LinkedIn Layout. 

I also added a bunch of functionality: 

1) I made "Smart Addresses" that will now work with Google's new geoencoder rules. Essentially, it checks to see if it's an american address, if it isn't it has special rules to make sure things are mapped correctly. I parse the address based on the number of commas that appear in the address field. 

2) You can now add all of your websites listed to the lead contact. But if there is a website that is listed with the type (Company Website), it'll add that to the main lead URL. 

3) I added support to parse twitter handles and skype handles, but I didn't activate the skype handles because of the bug I'm about to email engineering about. But, I left in the functionality for a later release. 

4) Support for descriptions of all sizes because their new layout truncates descriptions weirdly. 

That about covers it. Let me know what you think!

### CODE REVIEW

File: js/contentscript.js [hunk 1]
```diff
@@ -71,6 +71,22 @@ function getQueryParam(href, query) {
     }
 }
 
+function isInUS(location)
+{
+
+//Is a location in the US? Used to choose State Vs. Country in Address fields with 1 comma on LinkedIn.
+
+ array = ['Alabama','Alaska','American Samoa','Arizona','Arkansas','California','Colorado','Connecticut','Delaware','District of Columbia','Federated States of Micronesia','Florida','Georgia','Guam','Hawaii','Idaho','Illinois','Indiana','Iowa','Kansas','Kentucky','Louisiana','Maine','Marshall Islands','Maryland','Massachusetts','Michigan','Minnesota','Mississippi','Missouri','Montana','Nebraska','Nevada','New Hampshire','New Jersey','New Mexico','New York','North Carolina','North Dakota','Northern Mariana Islands','Ohio','Oklahoma','Oregon','Palau','Pennsylvania','Puerto Rico','Rhode Island','South Carolina','South Dakota','Tennessee','Texas','Utah','Vermont','Virgin Island','Virginia','Washington','West Virginia','Wisconsin','Wyoming'];
```
Comment (anemitz) [inline js/contentscript.js:79]: var

File: js/contentscript.js [hunk 2]
```diff
@@ -139,79 +155,163 @@ function handleFacebook() {
...
+    //         var platform = str_open_paren.split(')')[0].trim();
+    //         if(platform === "Skype")
+    //         {
+    //             var skype_handle = handle.split('(')[0].trim();
+    //             //lead.contacts[0].urls.push({ 'url': "skype://" + skype_handle, 'type': 'url' });
+    //         }
+            
+    //     }
+    // }
+
```
Comment (anemitz) [inline js/contentscript.js:314]: if not used I would remove this.

File: js/contentscript.js [hunk 3]
```diff
@@ -139,79 +155,163 @@ function handleFacebook() {
...
     if (twitter) {
-        lead.contacts[0].urls.push({ url: 'https://twitter.com/' + twitter, type: 'url' });
+        lead.contacts[0].urls.push({ 'url': twitter, 'type': 'url' });
     }
-    if (profile) {
-        lead.contacts[0].urls.push({ url: profile, type: 'url' });
+
+    //push website url to lead
+    if (websites) {
+        for (var i=0,  len=website_urls.length; i < len; i++)
```
Comment (anemitz) [inline js/contentscript.js:286]: use consistent spacing. like var x = y instead of x=y.

File: js/contentscript.js [hunk 4]
```diff
@@ -139,79 +155,163 @@ function handleFacebook() {
...
+
+    //Get Email Address From Contact Details
+    var email = $("section.pv-contact-info__contact-type.ci-email").find("span").last().text().trim();
+
+    //parse addresses on linkedin
+    var address_full = document.getElementsByClassName("pv-top-card-section__location Sans-15px-black-55% mb1 inline-block")[0].innerHTML|| "" ;
+    address_full = address_full.replace('Area', '');
+    address_full = address_full.replace('Greater', '');
+    address_full = address_full.trim();
+    var city = "", state = "", country = "", address_1 = "";
```
Comment (anemitz) [inline js/contentscript.js:208]: var city = state = country = address_1 = "";

**Comment by eengoron** [comment]
I've made every change except for the var city = state = country = address_1 ="" because grunt won't let me build it like that due to a warning that it's leaking a variable. I forced it through, but it's probably not ideal.

**Comment by anemitz** [comment]
ah, yeah could be. i'm not JS certified.

**Comment by anemitz** [comment]
Does this work for sales navigator or just the new LI layout?

Is it backwards compatible with the old one (if still in use)?

**Comment by eengoron** [comment]
Just the new LI Layout for now. 

Haven't touched sales navigator. But, the new layout thing was more complained about

And the old layout is only in use on IE, so it wouldn't matter. But, no it isn't backwards compatible. It's just completely updated.

**Comment by eengoron** [comment]
I'm going to open a new pull request with the changes you mentioned.

**Pull request closed without merging**